test/e2e2/fixture/fixture.go

```diff
@@ -85,8 +85,31 @@ func EnsureDeletion(ctx context.Context, kclient KubeClient, obj KubeObject) err
 		return err
 	}
 
+	// Wait for the object to be deleted  for 60 seconds
+	// - Primarily this will be waiting for the finalizer to be removed, so that the object is deleted
 	key := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
-	for count := 0; count < 120; count++ {
+	for count := 0; count < 60; count++ {
+		err := kclient.Get(ctx, key, obj, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return nil
+		} else if err == nil {
+			time.Sleep(1 * time.Second)
+		} else {
+			return err
+		}
+	}
+
```

Comment:
Not strictly related to the PR, but this makes the deletion function slightly more robust by removing finalizers if we're not able to delete within 60 seconds.